File: src/main/java/io/github/pavelshe11/messengermicro/store/repositories/ParticipantTypeRepository.java
package io.github.pavelshe11.messengermicro.store.repositories;

import io.github.pavelshe11.messengermicro.store.entities.ParticipantTypeEntity;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface ParticipantTypeRepository extends JpaRepository<ParticipantTypeEntity, UUID> {
    Optional<ParticipantTypeEntity> findByTypeName(String type);
}

File: src/main/java/io/github/pavelshe11/messengermicro/store/repositories/ParticipantTypeRepository.kt
package io.github.pavelshe11.messengermicro.store.repositories

import io.github.pavelshe11.messengermicro.store.entities.ParticipantTypeEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface ParticipantTypeRepository : JpaRepository<ParticipantTypeEntity?, UUID?> {
    fun findByTypeName(type: String): ParticipantTypeEntity?
}
